src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts

```diff
@@ -0,0 +1,172 @@
+        const proceedingDetailsList: ProceedingsOrderDataInterface[] = [];
+        const orderDetails = UserCase['otherProceedings']?.['order'][`${order}s`];
+        console.log(`${order}`);
+        orderDetails.forEach(nestedOrder => {
+          const orderDocumentDetails: Document = {
+            document_url: nestedOrder?.orderDocument?.url,
+            document_filename: nestedOrder?.orderDocument?.filename,
+            document_binary_url: nestedOrder?.orderDocument?.binaryUrl,
+          };
+          if (nestedOrder.orderDocument) {
+            const otherDetailsInfo: OtherProceedingDetails = {
+              orderDetail: nestedOrder.orderDetail,
+              caseNo: nestedOrder.caseNo,
+              currentOrder: nestedOrder.currentOrder,
+              orderCopy: nestedOrder.orderCopy,
+              orderDate: getLocalDate(nestedOrder.orderDate),
+              orderEndDate: getLocalDate(nestedOrder.orderEndDate),
+              orderDocument: orderDocumentDetails,
+            };
+            const proceedingDetailsInfo: ProceedingsOrderDataInterface = {
+              id: '',
+              value: otherDetailsInfo,
+            };
+            proceedingDetailsList.push(proceedingDetailsInfo);
+          } else {
+            const otherDetailsInfo: OtherProceedingDetails = {
+              orderDetail: nestedOrder.orderDetail,
+              caseNo: nestedOrder.caseNo,
+              currentOrder: nestedOrder.currentOrder,
+              orderCopy: nestedOrder.orderCopy,
+              orderDate: getLocalDate(nestedOrder.orderDate),
+              orderEndDate: getLocalDate(nestedOrder.orderEndDate),
+            };
+            const proceedingDetailsInfo: ProceedingsOrderDataInterface = {
+              id: '',
+              value: otherDetailsInfo,
+            };
+            proceedingDetailsList.push(proceedingDetailsInfo);
+          }
+        });
+        const proceedings: Proceedings = {
+          orderType: `${order}`,
+          proceedingDetails: proceedingDetailsList,
+        };
+        const proceedingData: ProceedingDetailsData = {
+          id: '',
+          value: proceedings,
+        };
+        proceedingDetails.push(proceedingData);
+      }
+      currentOrPreviousProceedings.haveChildrenBeenInvolvedInCourtCase = req.session.userCase.proceedingsStart;
+      currentOrPreviousProceedings.courtOrderMadeForProtection = req.session.userCase.proceedingsStartOrder;
+      currentOrPreviousProceedings.proceedingsList = proceedingDetails;
+    });
+    respondent.value.response.currentOrPreviousProceedings = currentOrPreviousProceedings;
+  } else {
+    respondent.value.response.currentOrPreviousProceedings = currentOrPreviousProceedings;
+  }
+
+  return respondent;
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const getProceedingDetails = (respondent: Respondent, req: AppRequest): Partial<CaseWithId> => {
```

Comment:
This method is creating side effects.

Reply:
made changes and used local variable as discussed